deployments/google_services.go

```diff
@@ -186,15 +188,15 @@ func (k GoogleServices) Deploy(c *kubernetes.Cluster, ui *termui.UI, options kub
 
 	ui.Note().KeeplineUnder(1).Msg("Deploying GoogleServices...")
 
-	err = k.apply(c, ui, options, false)
+	err = k.apply(ctx, c, ui, options, false)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (k GoogleServices) Upgrade(c *kubernetes.Cluster, ui *termui.UI, options kubernetes.InstallationOptions) error {
+func (k GoogleServices) Upgrade(ctx context.Context, c *kubernetes.Cluster, ui *termui.UI, options kubernetes.InstallationOptions) error {
 	_, err := c.Kubectl.CoreV1().Namespaces().Get(
 		context.Background(),
```

Comment:
use ctx here?

Reply:
I fixed all the remaining places and removed TODO() and Background() contexts.